cli/runner/cmd/requests.go

```diff
@@ -0,0 +1,40 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+//file to contain helper functions for different commands
+
+// Simple struct to hold JSON vals from API call
+type Langs struct {
+	Languages []string `json:"languages"`
+	Error     bool
+}
+
+func getLangListJSON(server string, endpoint string) (*Langs, error) {
+	var jsonLangs Langs
+	resp, err := http.Get(server + endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	//should I just avoid the status code check and just have one single error check here?
+	if resp.StatusCode > 299 {
+		return nil, fmt.Errorf("Response failed with status code: %d\n", resp.StatusCode)
+	} else if err != nil {
+		return nil, err
+	}
```

Comment:
I missed commenting this before, sorry. You're right to have both checks here since [http.Get](https://pkg.go.dev/net/http#Client.Get) has like a few failure cases:

1. Error
    1. `err != nil`: there is a `*url.Error`
    2. `err != nil` and `resp.StatusCode` is not 2xx: server returned an error response (don' worry about handling these cases specifically)
        1. 4xx: invalid request
        2. 5xx: server failure, timeout, etc
3. Ok
    1. `err != nil` and `resp.StatusCode` is 2xx (it could be 1xx but idk if anyone uses these codes...)

I just have one final nit pick: could we move the `resp.StatusCode` check before `io.ReadAll`? Since those two checks are handling possible "errors" for the `http.Get` call. It's not wrong to have it where it is but I personally think it makes more sense right after `http.Get`.

Thanks!